lib/vm.py

```diff
@@ -123,6 +136,7 @@ def wait_for_os_booted(self):
         # waiting for the IP:
         # - allows to make sure the OS actually started (on VMs that have the management agent)
         # - allows to store the IP for future use in the VM object
+        wait_for(self.try_get_ip_xenstore, "Wait for Xenstore VM IP")
         wait_for(self.try_get_and_store_ip, "Wait for VM IP")
```

Comment:
Does this solve the issue? IIUC the IP does appear in the xenstore but not seen by XAPI so the 2nd wait would fail anyway right?

Reply:
The goal is not to solve the issue, but to make it more visible when it happens.